vNext/Directory.Build.props

```diff
@@ -1,21 +1,30 @@
 <Project>
+  <PropertyGroup>
+    <VersionPrefix>2.0.0</VersionPrefix>
+    <VersionSuffix>preview2</VersionSuffix>
+  </PropertyGroup>
+
   <PropertyGroup Label="NuGet">
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <IncludeSource>true</IncludeSource>
     <IsPackable>false</IsPackable>
 
     <Authors>Microsoft</Authors>
-    <Copyright>Copyright Microsoft 2015</Copyright>
+    <Product>Microsoft Orleans</Product>
+    <Copyright>Copyright (c) Microsoft Corporation 2017</Copyright>
     <LicenseUrl>https://github.com/dotnet/Orleans#license</LicenseUrl>
     <ProjectUrl>https://github.com/dotnet/Orleans</ProjectUrl>
     <IconUrl>https://raw.githubusercontent.com/dotnet/orleans/gh-pages/assets/logo_128.png</IconUrl>
     <Tags>Orleans Cloud-Computing Actor-Model Actors Distributed-Systems C# .NET</Tags>
     <ReleaseNotes></ReleaseNotes>
     <RepositoryUrl>https://github.com/dotnet/Orleans</RepositoryUrl>
     <RepositoryType>git</RepositoryType>
+  </PropertyGroup>
 
-    <!-- Temporary properties until NuGet bug will be fixed: https://github.com/NuGet/Home/issues/4508 -->
-    <!-- Keep in sync with Build\Version.txt -->
-    <VersionPrefix>2.0.0.0</VersionPrefix>
-    <VersionSuffix>preview1</VersionSuffix>
+  <!-- For Debug builds generated a date/time dependent version suffix -->
+  <PropertyGroup Condition=" '$(Configuration)'=='Debug' ">
+    <VersionSuffix Condition=" '$(VersionSuffix)'=='' ">dev</VersionSuffix>
+    <VersionSuffix>$(VersionSuffix)-$([System.DateTime]::Now.ToString(yyyyMMddmmss))</VersionSuffix>
```

Comment:
You mean that it is evaluated on a per project basis, so datetime value can be different for 1 build?

I will look into it, can be a nice catch.